### PR TITLE
add dependency information to README

### LINKED ISSUE #38: Documentation Lacking
Can you please document which packages are required to build xedgewarp?

Do we need development files for libX11, X11 Protocol headers or what?

```
$ make
rm -f xedgewarp obj/xedgewarp.o obj/pointer.o obj/extensions.o obj/randr.o obj/event.o obj/util.o obj/fake_outputs.o
rm -f man/*.1 man/*.xml
rm -f test/GLOB*
gcc -D'__VERSION="heads/master-0-g04ca182"' -Iinclude -std=gnu99 -Wall -Wundef -Wshadow -o obj/xedgewarp.o -c src/xedgewarp.c
In file included from include/all.h:4:0,
                 from src/xedgewarp.c:2:
include/externals.h:13:36: fatal error: X11/extensions/XInput2.h: No such file or directory
 #include <X11/extensions/XInput2.h>
                                    ^
compilation terminated.
Makefile:33: recipe for target 'obj/xedgewarp.o' failed
make: *** [obj/xedgewarp.o] Error 1

```

### PR DESCRIPTION
**Pull request by nmschulte**
Fixes #38.  Please review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,13 +36,27 @@ xedgewarp is available in the AUR as [xedgewarp-git](https://aur.archlinux.org/p
...
+```
 git clone https://github.com/Airblader/xedgewarp
 cd xedgewarp
 make
 sudo make install
 ```
 
+### Dependencies
+
+xedgewarp depends upon varios X11 libraries, including:
```
Comment (Airblader) [inline README.md]: I don't understand the "bits and pieces". A list of depencies should be just that: a list of dependencies. Can you please rephrase it to leave out the "various" and "bits and pieces" part and just state "xedgewarp has the following dependencies"?

File: README.md [hunk 2]
```diff
@@ -36,13 +36,27 @@ xedgewarp is available in the AUR as [xedgewarp-git](https://aur.archlinux.org/p
...
+ * bits from XCB for X11; [libxcb, libX11-xcb](https://xcb.freedesktop.org/)
+ * Input Extensions for X11; [libXi](https://xorg.freedesktop.org)
+ * Display Manager Control Protocol Library for X11; [libXdmcp](https://xorg.freedesktop.org)
```
Comment (Airblader) [inline README.md]: This is not a dependency of xedgewarp. What made you think so?

File: README.md [hunk 3]
```diff
@@ -36,13 +36,27 @@ xedgewarp is available in the AUR as [xedgewarp-git](https://aur.archlinux.org/p
...
+ * Miscellaneous Extension Library for X11; [libXext](https://xorg.freedesktop.org)
+ * bits from X11; [libx11](https://xorg.freedesktop.org)
+
+For Debian and apt/dpkg managed distributions:
```
Comment (Airblader) [inline README.md]: I don't like maintaining a long list of dependency packages in the README for all distributions. I'd prefer if somebody created a package instead which we can then link to – in general I'd like to avoid encouraging people to compile from source as this only brings up a lot of the same questions.

I'll happily link a package from the README (like I did for Manjaro / Archlinux).

**Review by Airblader — CHANGES_REQUESTED** [review]

**Comment by Airblader** [comment]
Don't worry about the test failures, they're quite obviously not your fault.

File: README.md (hunk 2 above)
Comment (nmschulte) [reply README.md]: I got lazy (and stupid, as I can see now);
```
$ ldd `which xedgewarp`
	linux-vdso.so.1 (0x00007ffe9bd6a000)
	libxcb-randr.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-randr.so.0 (0x00007ff2bee4a000)
	libxcb-util.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-util.so.0 (0x00007ff2bec43000)
	libX11-xcb.so.1 => /usr/lib/x86_64-linux-gnu/libX11-xcb.so.1 (0x00007ff2bea41000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007ff2be701000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007ff2be4d9000)
	libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007ff2be2c7000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ff2bdf29000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ff2bdd25000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007ff2bdb21000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007ff2bd91b000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007ff2bd709000)
	/lib64/ld-linux-x86-64.so.2 (0x000055f584126000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007ff2bd4f1000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007ff2bd2e9000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ff2bd0cc000)
```

~What are the dependencies, exactly?~ libxcb-randr.so, libxcb-util.so, libX11-xcb.so, libX11.so, libxcb.so, libXi.so, which I think shakes out like so:
XCB: libxcb-randr, libxcb-util, libxcb
X11: libX11-xcb, libX11, libXi

Comment (Airblader) [reply README.md]: Yes, your edited list looks better. :-)

File: Makefile [hunk 4]
```diff
@@ -4,9 +4,9 @@ IDIR = include
 ODIR = obj
 
 INSTALL = install
-PREFIX = /usr/bin
```
Comment (Airblader) [inline Makefile]: You need to rebase after the other PR has been merged since these changes are not part of this PR.

**Review by Airblader — APPROVED** [review]

**Comment by Airblader** [comment]
Thanks!
